projects/packages/sync/src/class-actions.php

```diff
@@ -99,6 +101,15 @@ public static function init() {
 			return;
 		}
 
+		// If dedicated Sync is enabled and this is a dedicated Sync request, no need to
+		// initialize Sync for cron jobs, set up listeners or set up a shut-down action
+		// for sending actions to WordPress.com.
+		// We only need to set up an init action for sending actions to WordPress.com and exit early.
+		if ( Settings::is_dedicated_sync_enabled() && Dedicated_Sender::is_dedicated_sync_request() ) {
```

Comment:
the above if statement mirrors the ones found in do_dedicated_sync_and_exit. It makes sense that we would check this here but why do we recheck in do_dedicated_sync_and_exit.

Reply:
`do_dedicated_sync_and_exit` is a public method so I'd prefer to be explicit in the checks there to prevent any potential bad usage in case it's used outside of the current flow.